packages/core/test/popover/popoverTests.tsx

```diff
@@ -165,6 +165,123 @@ describe("<Popover>", () => {
+            });
+
+            it("does not open popover on target focus when interactionKind is CLICK", () => {
+                assertPopoverOpenStateForInteractionKind(PopoverInteractionKind.CLICK, false);
+            });
+
+            it("does not open popover on target focus when interactionKind is CLICK_TARGET_ONLY", () => {
+                assertPopoverOpenStateForInteractionKind(PopoverInteractionKind.CLICK_TARGET_ONLY, false);
+            });
+
+            it.skip("closes popover on target blur if autoFocus={false}", () => {
+                // TODO (clewis): This is really tricky to test given the setTimeout in the onBlur implementation.
+            });
+
+            it("popover remains open after target focus if autoFocus={true}", () => {
+                wrapper = renderPopover({
+                    autoFocus: true,
+                    inline: false,
+                    interactionKind: PopoverInteractionKind.HOVER,
+                });
+                const targetElement = wrapper.find(`.${Classes.POPOVER_TARGET}`);
+                targetElement.simulate("focus");
+                targetElement.simulate("blur");
+                assert.isTrue(wrapper.state("isOpen"));
+            });
+        });
+
+        describe("if false", () => {
+            it("does not add tabindex to target's child node when interactionKind is HOVER", () => {
+                assertPopoverTargetTabIndex(PopoverInteractionKind.HOVER, false, false);
+            });
+
+            it("does not add tabindex to target's child node when interactionKind is HOVER_TARGET_ONLY", () => {
+                assertPopoverTargetTabIndex(PopoverInteractionKind.HOVER_TARGET_ONLY, false, false);
+            });
+
+            it("does not add tabindex to target's child node when interactionKind is CLICK", () => {
+                assertPopoverTargetTabIndex(PopoverInteractionKind.CLICK, false, false);
+            });
+
+            it("does not add tabindex to target's child node when interactionKind is CLICK_TARGET_ONLY", () => {
+                assertPopoverTargetTabIndex(PopoverInteractionKind.CLICK_TARGET_ONLY, false, false);
+            });
+
+            it("does not open popover on target focus when interactionKind is HOVER", () => {
+                assertPopoverOpenStateForInteractionKind(PopoverInteractionKind.HOVER, false, false);
+            });
+
+            it("does not open popover on target focus when interactionKind is HOVER_TARGET_ONLY", () => {
+                assertPopoverOpenStateForInteractionKind(PopoverInteractionKind.HOVER_TARGET_ONLY, false, false);
+            });
+
+            it("does not open popover on target focus when interactionKind is CLICK", () => {
+                assertPopoverOpenStateForInteractionKind(PopoverInteractionKind.CLICK, false, false);
+            });
+
+            it("does not open popover on target focus when interactionKind is CLICK_TARGET_ONLY", () => {
```

Comment:
you want a fun refactor? generate these `it` titles in the test function itself. will make the test setup more legible.

```ts
function assertSomething(interactionKind: PopoverInteractionKind, isOpen: boolean, openOnTargetFocus?: boolean) {
    it(`${isOpen ? "does" : "does not"} open popover on target focus when interactionKind is ${etc}`) {
```